flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/resource/ResourceManager.java

```diff
@@ -54,7 +54,7 @@ public class ResourceManager implements Closeable {
 
     private final Path localResourceDir;
     private final Map<ResourceUri, URL> resourceInfos;
-    private final MutableURLClassLoader userClassLoader;
+    private MutableURLClassLoader userClassLoader;
 
     public ResourceManager(Configuration config, MutableURLClassLoader userClassLoader) {
         this.localResourceDir =
```

Comment:
In the gateway side, I am prone to control the directory by the gateway itself. For example, I want to create the directory with `sql-gateway-1.14-<SessionHandle>`. Could you share some thoughts how I can do this?